### PR TITLE
fix: patch SO put strikes

### PR DESCRIPTION
**Pull request by diegofigs**
## Description
PR patches SO tab to use helper method for calculating strike dependent on direction

### CODE REVIEW

File: app/src/hooks/useStakingOptions.ts [hunk 1]
```diff
@@ -156,3 +165,8 @@ export interface SoParams {
   maxSettlement: number;
   maxFees: number;
 }
+
+export function calculateStrikeAtomsPerBaseToken(data: { isUpside: boolean; mint: PublicKey; strike: number }) {
```
Comment (brittcyr) [inline app/src/hooks/useStakingOptions.ts:169]: calculateStrikeQuoteAtomsPerBaseToken

File: app/src/hooks/useStakingOptions.ts [hunk 2]
```diff
@@ -73,13 +73,22 @@ async function fetchData(provider: AnchorProvider): Promise<SoParams[]> {
         continue;
       }
 
+      const quote = quoteMint.toString();
+      const isUpside =
```
Comment (brittcyr) [inline app/src/hooks/useStakingOptions.ts:77]: make a helper function for this

File: app/src/hooks/useStakingOptions.ts (hunk 1 above)
Comment (brittcyr) [inline app/src/hooks/useStakingOptions.ts:169]: be explicit which mint it is.

Comment (brittcyr) [inline app/src/hooks/useStakingOptions.ts:169]: include units on strike var name in the input

File: app/src/hooks/useStakingOptions.ts [hunk 3]
```diff
@@ -73,13 +73,22 @@ async function fetchData(provider: AnchorProvider): Promise<SoParams[]> {
...
         name: soName,
         authority: new PublicKey(authority),
         expiration: new Date(Number(optionExpiration) * 1_000).toDateString().split(' ').slice(1).join(' '),
         expirationInt: Number(optionExpiration),
-        strike: roundedStrike,
+        strike: calculateStrikeAtomsPerBaseToken({
+          isUpside,
+          mint: isUpside ? quoteMint : baseMint,
+          strike: strike.toNumber(),
+        }).toString(),
```
Comment (brittcyr) [inline app/src/hooks/useStakingOptions.ts:91]: why a string? seems like it is something that should be a number.
or are we trimming to correct decimals here so we dont format later?

**Review by brittcyr — APPROVED** [review]

**Comment by DonDuala** [comment]
Strikes have the right digits but are off by a factor of 100, likely from a missing baseAtoms/quoteAtoms = 10^8 / 10^6

image

**Comment by DonDuala** [comment]
> Strikes have the right digits but are off by a factor of 100, likely from a missing baseAtoms/quoteAtoms = 10^8 / 10^6
> 
> image

SOL off by 1000x since 10^9 atoms

**Pull request merged**